build.yaml

```diff
@@ -1,8 +1,27 @@
+schedules:
+  commit:
+    schedule: per_commit
+    matrix:
+      exclude:
+        # Exclude all 0.12 and 4 builds.
+        - nodejs: ['0.12', '4']
+        # Only build with latest for 0.10
+        - nodejs: '0.10'
+          cassandra: ['1.2', '2.0', '2.1', '2.2', '3.0']
+        # Only build with 1.2 and latest for 0.10
+        - nodejs: '6'
+          cassandra: ['2.0', '2.1', '2.2', '3.0']
+  nightly:
+    schedule: nightly
+    branches:
+      # regex matches primary branch format (2.1, 3.x, 3.0.x, 3.1.x, master, etc).
+      include: ["/((\\d+(\\.[\\dx]+)+)|(master))/"]
```

Comment:
Yeah, I think there is definitely some redundancy there, primarily in surrounding a lot in parentheses where it's not needed (don't really need capture groups).   The escaping might not be needed either (it is for java but not for ruby iirc).  I'll tweak this.